### PR TITLE
More fixes to supporting util providers

### PR DESCRIPTION
**Pull request by shefty**

### CODE REVIEW

**Comment by shefty** [comment]
Fixes #2892

**Comment by shefty** [comment]
@gladkovdmitry17 - this reverts your previous changes to fi_getinfo.

**Comment by gladkovdmitry17** [comment]
Thank you for the information. But I'm afraid that it can lead to broken initialization of util providers in the step of core provider initialization. Since the `prov_name` string in `fabric_attr` are contain `util_name` the checking of `core_name` will be never reached. Please,  check my comment. Guess it can help in this case.

File: src/fabric.c [hunk 1]
```diff
@@ -610,29 +610,22 @@ int DEFAULT_SYMVER_PRE(fi_getinfo)(uint32_t version, const char *node,
...
+		}
 
-			if (util_len && util_name &&
-			    strncasecmp(util_name, prov->provider->name, util_len)) {
-				FI_INFO(&core_prov, FI_LOG_CORE,
-					"Want util %s, skipping %s\n",
-					util_name, prov->provider->name);
+		if (util_len && util_name) {
+			assert(!(flags & OFI_CORE_PROV_ONLY));
+			if (strncasecmp(util_name, prov->provider->name, util_len))
```
Comment (gladkovdmitry17) [inline src/fabric.c]: I guess that combined check can help us to reach comparing of `core_name` with `provider->name` that is below in `else-if` statement 
I propose the following:
`if (util_len && util_name &&
     strncasecmp(util_name, prov->provider->name, util_len)) {`

**Comment by shefty** [comment]
If the prov_name contains a util_name, then getinfo call must be routed to the utility provider.  Otherwise the proper layering cannot be enforced.  For example, if prov_name = "ofi-rxm;sockets", and we route to sockets, it is capable of responding to the query.  This results in the app using only the sockets provider, and not through the rxm provider.
To support the layering, we route to the util provider first.  The util provider then strips the util name from the prov_name (i.e. "ofi-rxm;sockets" -> "socket").  This allows the subsequent getinfo call to route to the correct core provider.  This is what the code currently does when passing through a util provider.

**Comment by a-ilango** [comment]
@shefty thanks for the fixes. I have verified that this works against https://github.com/ofiwg/fabtests/pull/638

**Comment by shefty** [comment]
We can continue this discussion, but merging this, since it's blocking other updates.

**Pull request merged**